model/src/pyrenew/metaclass.py

```diff
@@ -412,3 +412,76 @@ def plot_posterior(
             draws_col=draws_col,
             obs_col=obs_col,
         )
+
+    def posterior_predictive(
+        self,
+        rng_key: ArrayLike | None = None,
+        *args,
+        guide: Callable | None = None,
+        params: dict | None = None,
+        num_samples: int | None = None,
+        return_sites: list[str] | None = None,
+        infer_discrete: bool = False,
+        parallel: bool = False,
+        batch_ndims: int | None = None,
+        exclude_deterministic: bool = True,
+        **kwargs,
+    ) -> dict:
```

Comment:
To check my understanding: these keyword arguments need to be enumerated explicitly rather than lumped under `**kwargs`, even though we just use the `Predictive` constructor defaults because `Predictive`'s `__init__()` method does not support arbitrary `**kwargs` but its `__call__()` method __does__, and we want to enable the user to specify __both__ keyword arguments for the `Predictive` constructor and keyword arguments for the `__call__()`.

Reply:
Yes. One downside is that the function signature for the Predictive constructor could change underneath us.

Reply:
Got it; thanks. We can think about alternative approaches in the future, but I think this is fine for now.

Reply:
I like it.

Reply:
I like that too. `**kwargs` to the `__call__()` but a dict of `predictive_args` if you want to use non-default values in the `Predictive` constructor

Reply:
Addressed in https://github.com/CDCgov/multisignal-epi-inference/pull/218/commits/a7f5557bacd4cea30ebf8b41da7a3e6d47d028bc.